src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -356,7 +359,7 @@ module internal Commitments =
                         )
                     localHtlcSigsAndHTLCTxs |> List.map(fst), localHtlcSigsAndHTLCTxs |> List.map(snd) |> Seq.cast<IHTLCTx> |> List.ofSeq
 
-                let remoteHTLCPubKey = Generators.derivePubKey ctx (cm.RemoteParams.HTLCBasePoint) localPerCommitmentPoint.PubKey
+                let remoteHTLCPubKey = Generators.derivePubKey ctx (remoteChannelKeys.HTLCBasePubKey) localPerCommitmentPoint.PubKey
```

Comment:
nit: let's remove the parenthesis now that you're changing this line, they are not needed